bundles/org.openhab.binding.panamaxfurman/src/main/java/org/openhab/binding/panamaxfurman/internal/PanamaxFurmanAbstractHandler.java

```diff
@@ -0,0 +1,250 @@
+import org.openhab.binding.panamaxfurman.internal.transport.PanmaxFurmanConnector;
+import org.openhab.binding.panamaxfurman.internal.util.TimeInterval;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PanamaxFurmanHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Dave Badia - Initial contribution
+ */
+@NonNullByDefault
+public abstract class PanamaxFurmanAbstractHandler extends BaseThingHandler
+        implements PanamaxFurmanCommunicationEventListener {
+    private final Logger logger = LoggerFactory.getLogger(PanamaxFurmanAbstractHandler.class);
+
+    private static final long AVOID_DUPLICATE_REQUESTS_WITHIN_MILLIS = TimeUnit.SECONDS.toMillis(3);
+    private static final Map<String, Long> REQUEST_LAST_SENT_AT = new ConcurrentHashMap<>();
+    private final ProtocolMapper protocolMapper;
+    /**
+     * Nullable since PanmaxFurmanConnector constructor will throw an exception if it can't connect
+     */
+    private @Nullable PanmaxFurmanConnector connector;
+    private @Nullable ScheduledFuture<?> backgroundJob;
+
+    public PanamaxFurmanAbstractHandler(Thing thing, ProtocolMapper protocolMapper) {
+        super(thing);
+        this.protocolMapper = protocolMapper;
+    }
+
+    /**
+     * Request the implementation to build a connector for the protocol
+     *
+     * @return the connector
+     */
+    protected abstract @Nullable PanmaxFurmanConnector buildConnector(Configuration genericConfig);
+
+    /**
+     * The name of this device connection
+     *
+     * @return a description of this connection for logging purposes
+     */
+    protected abstract String getConnectionName();
+
+    /**
+     * The interval at which the device should be polled for status updates. Will vary depending on whether the
+     * underlying protocol supports push updates or not
+     *
+     * @return the interval at which update polling should take place
+     */
+    protected abstract TimeInterval pollInterval();
+
+    @Override
+    public void initialize() {
+        try {
+            updateStatus(ThingStatus.UNKNOWN);
+            this.connector = buildConnector(getConfig());
+            logger.debug("starting scheduled status now then repeating {}", getConnectionName());
+            // Request ID info immediately
+            if (connector != null) {
+                connector.sendRequestToDevice(protocolMapper.buildQueryString(PowerConditionerChannel.ID, null));
```

Comment:
Init should return fast. This query can take time. Can you make this async ?

Reply:
Got it, will do

Reply:
Can you mark it fixed once you have your changes pushed? Easier for me to keep track of changes. (at time of writing, this was marked resolved, but i don't see a change to the code)

Reply:
Oh sorry about that.  i made the code change but want to test it before committing

Reply:
That's oke, upate the comments once changes are pushed and you want me to look at it again.